src/components/Admin/ManageSensors/DeleteSensorModal.tsx

```diff
@@ -0,0 +1,208 @@
+      setIsLoading(true);
+
+      const deletionDocRef = firestore.collection('deletion').doc('todo');
+
+      // Get current mapping or set to empty
+      deletionDocRef
+        .get()
+        .then(deleteDoc => {
+          const newDeletionMap =
+            deleteDoc.data()?.deletionMap ?? Object.create(null);
+
+          newDeletionMap[sensorDocId] = new Date();
+
+          deletionDocRef
+            .update({deletionMap: newDeletionMap})
+            .then(() => {
+              firestore.collection('sensors').doc(sensorDocId).delete();
+              setIsLoading(false);
+            })
+            .then(handleClose);
+        })
+        .catch(() => setError('deleteSensor.deleteSensorError'))
+        .finally(() => setIsLoading(false));
+    }
+  }
+
+  return (
+    <Box>
+      <Button colorScheme="red" onClick={onOpen}>
+        {t('deleteSensor.delete')}
+      </Button>
+      <Modal isOpen={isOpen} onClose={handleClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>{t('deleteSensor.delete')}</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <Box>
+              <Text>{t('deleteSensor.whichSensorToDelete')}</Text>
+              <SensorInput
+                sensors={sensors}
+                docId={sensorDocId}
+                setDocId={setSensorDocId}
+                setPurpleAirId={setPurpleAirId}
+              />
+            </Box>
+            <Box marginTop={2}>
+              <Box>
+                <LabelValue
+                  label={t('sensors.purpleAirId')}
+                  value={purpleAirId}
+                />
+              </Box>
+              <Checkbox
+                isChecked={confirmDownload}
+                onChange={() => setConfirmDownload(!confirmDownload)}
+              >
+                {t('deleteSensor.confirmDownload')}
+              </Checkbox>
+              <Checkbox
+                isChecked={acknowledgeDeletion}
+                onChange={() => setAcknowledgeDeletion(!acknowledgeDeletion)}
+              >
+                {t('deleteSensor.acknowledgeDelete')}
+              </Checkbox>
+              <Checkbox
+                isChecked={acknowledgePermanent}
+                onChange={() => setAcknowledgePermanent(!acknowledgePermanent)}
+              >
+                {t('deleteSensor.cannotBeUndone')}
+              </Checkbox>
+              <FormControl isRequired isInvalid={error !== ''} marginTop={4}>
+                <FormLabel>{t('deleteSensor.confirmPurpleAirId')}</FormLabel>
+                <Input
+                  placeholder="30971"
+                  size="md"
+                  onChange={event => {
+                    setConfirmPurpleAirId(event.target.value);
+                    setError('');
+                  }}
+                  value={confirmPurpleAirId}
+                />
+                <FormErrorMessage>{error}</FormErrorMessage>
+              </FormControl>
+              <Button
```

Comment:
Should we add a left icon of FaTrash? The one on the delete old modal has it

Reply:
I agree, I added the same code you had in DeleteOldDataModal